packages/drivers/odsp-driver/src/fetchSnapshot.ts

```diff
@@ -263,3 +327,62 @@ export function evalBlobsAndTrees(snapshot: IOdspSnapshot) {
     }
     return { numTrees, numBlobs, encodedBlobsSize, decodedBlobsSize };
 }
+
+export function isRedeemSharingLinkError(odspResolvedUrl: IOdspResolvedUrl, error: any) {
+    const errorType = error.errorType;
+    if (odspResolvedUrl.sharingLinkToRedeem !== undefined
+        && (errorType === DriverErrorType.authorizationError
+        || errorType === DriverErrorType.fileNotFoundOrAccessDeniedError)) {
+        return true;
+    }
+    return false;
+}
+
+function encodeShareUrl(url: string): string | undefined {
+    if (!url) {
+      return undefined;
+    }
+
+    /**
+     * Encode the url to accepted format by Sharepoint
+     * https://docs.microsoft.com/en-us/onedrive/developer/rest-api/api/shares_get
+     */
+    let encodedUrl = b64EncodeUnicode(url);
+    if (!encodedUrl) {
+      return undefined;
+    }
+
+    encodedUrl = encodedUrl
+      .replace(/=+$/g, "")
+      .replace(/\//g, "_")
+      .replace(/\+/g, "-");
+    encodedUrl = "u!".concat(encodedUrl);
+
+    return encodedUrl;
+}
+
+function b64EncodeUnicode(str: string): string | undefined {
+    if (!str || typeof str !== "string") {
+      return undefined;
+    }
+
+    // first we use encodeURIComponent to get percent-encoded UTF-8,
+    // then we convert the percent encodings into raw bytes which
+    // can be fed into btoa.
+    // Latin characters like # which are also special characters are not encoded.
+    // Hence we manually encode them with encodeHash method - This is a requirement from Vroom APIs
+    return btoa(
```

Comment:
Can you please describe what this sequence is supposed to do, not how it does it?
How is it different from what fromUtf8ToBase64(encodeURIComponent()) does?
Also you description suggests that # is one of the special characters, but there is only it getting handled.
If this sequence is correct, it would be great to 
- have a reference to some SPO doc describing expectations here.
- inline encodeHash (there is already call to .replace, you can simply chain them, not clear why one of the is a separate function).

Reply:
Removed this entire code and replaced it with much simpler. I was trying to use this "fromUtf8ToBase64(encodeURIComponent())" but was not able to get the correctly encoded one. Digging into it more I got it correct with using encodeUrl instead of encodeUriComponent and found there is difference in both. So replaced this code.